PubSub/src/Connection/Gapic.php

```diff
@@ -45,10 +45,11 @@
 use Google\Protobuf\Timestamp;
 
 /**
- * Implementation of the
+ * Implementation of the GAPIC connection class which uses both
+ * REST and GRPC transports via
  * [Google Pub/Sub gRPC API](https://cloud.google.com/pubsub/docs/reference/rpc/).
  */
-class Grpc implements ConnectionInterface
+class Gapic implements ConnectionInterface
```

Comment:
I see why this is done - renaming the `Grpc` connection is a clever way to keep the code DRY. However, what I would do instead is either make a `GapicTrait` which contains all these methods, and use it in both `Gapic` and `Grpc` connection classes, or have them extend one another and override the `getGaxConfig` method.

This is because I think we should mark these connection classes `@deprecated` instead of removing them entirely (until we cut a major version).